app/controllers/concerns/unconfirmed_user_concern.rb

```diff
@@ -25,11 +25,23 @@ def track_user_already_confirmed_event
       errors: { email: [t('errors.messages.already_confirmed')] },
       user_id: @user.uuid,
     )
+    irs_attempts_api_tracker.user_registration_email_confirmation(
+      email: @email_address.email,
+      success: false,
+      errors: { email: [t('errors.messages.already_confirmed')] },
+      error_details: { email: [t('errors.messages.already_confirmed')] },
```

Comment:
can we extract `errors` to a variable and share this with above?

Also looks like we don't need `error_details` at all

Reply:
This is kind of my fault. In the schema we actually called the parameter `failure_reason`; we should use that and get rid of `errors` and `error_details`.

We specified this as an optional string, so I think we should just pass in something like this:
```suggestion
      failure_reason: 'Email already confirmed',
```

The i18n string looks like it's already a sentence fragment:
https://github.com/18F/identity-idp/blob/main/config/locales/errors/en.yml#L46

Since this is just going t an API for one consumer, I don't think we need to worry about internationalization, so I'd be tempted to just use a simple string like 'Email already confirmed', or maybe we make it "email_already_confirmed"? @zachmargolis, do you have an opinion?

Reply:
errors and error_details are populated from "form_response.rb" as hash and so we can add one new method (like "first_error_message" in form_response.rb) that returns us an error message as a string

Reply:
I like your thinking, but I think in this case we can skip that work. Since we're already in `track_user_already_confirmed_event`, we already know what the error is, so I think we can do something basic without looking at the error hash itself.

Reply:
Using anything out of `I18n` is not good because that means if two users in two different locales get the same error, they'll have different strings. I would recommend using a symbol/slug-like thing such as `email_already_confirmed`

Reply:
Yeah, but in stop_if_invalid_token we don't know what will be the error :(

Reply:
def failure_reason
    return if errors.blank?
    errors.map{|k, v| "#{k.to_s} #{v.first}" }.join(", ")
  end
  
  How do you think about this method to be added in form_response.rb that will convert
  
  :errors=>{:confirmation_token=>["has expired, please request a new one"]},
  to
  :errors=>"confirmation_token has expired, please request a new one",

Reply:
I didn't comment on the above earlier because it came up in chat, but I like this approach. I didn't want us to have to invest a ton in error-handling code for what feels like an edge case, but I like that you were able to implement this simply.